ex03.exs

```diff
@@ -49,18 +49,50 @@ defmodule Ex03 do
      it contains no conditional logic:  3
      it is nicely structured            7
   """
+  #pmap(1..10, 1, &(&1+1)
 
   def pmap(collection, process_count, function) do
-    # your code goes here
-    # I'm hoping to see a simple pipeline as the body of this function...
+      #divideCollection will divide collection into process_count chunks
+      #main_function will spawn threads with the dividedCollection chunks in each thread
+      #thread_flattener will take the dividedCollection and flatten it to a single collection
+      divideCollection(collection,process_count) |> main_function(function) |> thread_flattener()
   end
 
   # and here...
+  def divideCollection(collection, process_count) do
+  #divides collection into chunks based on process_count
+  collection =  Enum.chunk_every(collection, div(Enum.count(collection), process_count),div(Enum.count(collection), process_count),[])
```

Comment:
Two problems here:

First, the variable `collection` is never used, which generates a warning. Warnings should always be resolved: in my experience they are often errors.

# -1

Second, the `div` call returns an incorrect value for the chunk size: for 10, 4, it returns 2, but it should be 3.

# -2